File: 104-MaximumDepthOfBinaryTree/104-MaximumDepthOfBinaryTree.js
// Last updated: 29/05/2025, 21:42:51
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxDepth = function(root) {
    if (root === null ) {
        return 0
    }
    let depth = 0
    let queue = [root]
    let headQueueIndex = 0
    while (queue.length - headQueueIndex) {
        let levelSize = queue.length - headQueueIndex 
        for (let i = 0; i < levelSize; i++) {
            let item = queue[headQueueIndex]
            headQueueIndex += 1
            if (item.left) {
                queue.push(item.left)
            }
            if (item.right) {
                queue.push(item.right)
            }
        }
        depth += 1
    }
    return depth
};

File: 104-MaximumDepthOfBinaryTree/104-MaximumDepthOfBinaryTree.py
# Last updated: 29/05/2025, 21:42:49
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return 0
        max_depth_right = self.maxDepth(root.right)
        max_depth_left = self.maxDepth(root.left)
        return 1 + max(max_depth_right, max_depth_left)
